pkg/storage/etcd/etcd_helper.go

```diff
@@ -283,7 +304,11 @@ func (h *etcdHelper) extractObj(response *etcd.Response, inErr error, objPtr run
 		}
 		return "", nil, fmt.Errorf("unable to locate a value on the response: %#v", response)
 	}
-	body = node.Value
+
+	body, err = h.transformer.ReadString(node.Value)
+	if err != nil {
+		return body, nil, err
```

Comment:
I think we need a typed error here so that the API can report the issue back to the client which can decide how to proceed.